app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectViewModel.kt

```diff
@@ -161,24 +162,37 @@ class WalletConnectViewModel(application: Application) : AndroidViewModel(applic
-                    proxyRepository.getTransferCost(
-                        type = "update",
-                        amount = payload.amount.toBigInteger(),
-                        sender = walletConnectData.account!!.address,
-                        contractIndex = payload.address.index,
-                        contractSubindex = payload.address.subIndex,
-                        receiveName = payload.receiveName,
-                        parameter = payload.message,
-                        success = {
-                            walletConnectData.maxEnergy =
-                                if (payload.maxEnergy != 0L) payload.maxEnergy else payload.maxContractExecutionEnergy
-                            walletConnectData.cost = it.cost.toBigInteger()
-                            walletConnectData.energy = it.energy
+                    proxyRepository.getChainParameters(
+                        success = { chainParameters ->
+                            val maxEnergy: Long =
+                                if (payload.maxEnergy != null) {
+                                    // Use the legacy total value if provided.
+                                    payload.maxEnergy
+                                } else if (payload.maxContractExecutionEnergy != null) {
+                                    // Calculate the total value locally if only the execution energy provided.
+                                    // Max energy = contract execution energy + base transaction energy.
+                                    payload.maxContractExecutionEnergy +
+                                            TransactionCostCalculator.getBaseCostEnergy(
+                                                transactionSize = TransactionCostCalculator.getContractTransactionSize(
+                                                    receiveName = payload.receiveName,
+                                                    message = payload.message,
+                                                ),
+                                            )
+                                } else {
+                                    error("The account transaction payload must contain either maxEnergy or maxContractExecutionEnergy")
+                                }
+
+                            val cost = TransferCost(
+                                energy = maxEnergy,
+                                euroPerEnergy = chainParameters.euroPerEnergy,
+                                microGTUPerEuro = chainParameters.microGtuPerEuro,
+                            )
+
+                            walletConnectData.maxEnergy = cost.energy
+                            walletConnectData.cost = cost.cost
                             transactionFee.postValue(walletConnectData.cost)
                         },
-                        failure = {
-                            handleBackendError(it)
-                        }
+                        failure = ::handleBackendError
```

Comment:
What will happen in the UI in this case?

Previously it just proceeded with some dummy values, what happens now?

Reply:
Now it will show "Failed to connect"-style message and on UI there will be no fee and energy data, yet the value from the dApp (`maxEnergy` or `maxContractExecutionEnergy`) will remain in the payload hence will be used to create the transaction.